rustls/src/conn.rs

```diff
@@ -841,7 +841,7 @@ impl ConnectionCommon {
         // In the handshake case we don't have readable plaintext before the handshake has
         // completed, but also don't want to read if we still have sendable tls.
         self.received_plaintext.is_empty()
-            && !self.connection_was_cleanly_closed()
+            && !self.has_received_close_notify
```

Comment:
It doesn't seem like the existing of pending messages in the deframer should affect whether we want/need a read, after we've received a close notify from the peer (and this makes further changes easier because otherwise `wants_read()` depends on the deframer, which will stay in `ConnectionCommon`).